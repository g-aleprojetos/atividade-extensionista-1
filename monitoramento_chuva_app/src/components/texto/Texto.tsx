import React from 'react';
import textos from 'resources/textos';

import {TextoStyled, Props} from './Texto.styles';

export function Texto(props: Props) {
  return <TextoStyled {...props} />;
}

export function Texto3XSmall(props: Props) {
  return (
    <Texto tamanho={textos.tamanho.xxxsmall} alturaDeLinha={textos.altura.xxxsmall} {...props} />
  );
}

export function Texto2XSmall(props: Props) {
  return (
    <Texto tamanho={textos.tamanho.xxsmall} alturaDeLinha={textos.altura.xxsmall} {...props} />
  );
}

export function TextoXSmall(props: Props) {
  return <Texto tamanho={textos.tamanho.xsmall} alturaDeLinha={textos.altura.xsmall} {...props} />;
}

export function TextoSmall(props: Props) {
  return <Texto tamanho={textos.tamanho.small} alturaDeLinha={textos.altura.small} {...props} />;
}

export function TextoMedium(props: Props) {
  return <Texto tamanho={textos.tamanho.medium} alturaDeLinha={textos.altura.medium} {...props} />;
}
