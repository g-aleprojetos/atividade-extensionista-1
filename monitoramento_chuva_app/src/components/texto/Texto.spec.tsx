import React from 'react';
import {render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import * as S from './Texto';

describe('Texto', () => {
  describe('Renderização', () => {
    const setup = () => render(<S.Texto>Texto de teste</S.Texto>);
    test('DEVE renderizar o texto "Texto de teste"', () => {

      setup();      
      const result = screen.getByText('Texto de teste' );
      expect(result).toBeDefined();
    });

    test('Texto3XSmall DEVE ser igual ao snapshot', () => {
      const texto = renderer
        .create(<S.Texto3XSmall>Teste</S.Texto3XSmall>)
        .toJSON();
      expect(texto).toMatchSnapshot();
    });

    test('Texto2XSmall DEVE ser igual ao snapshot', () => {
      const texto = renderer
        .create(<S.Texto2XSmall>Teste</S.Texto2XSmall>)
        .toJSON();
      expect(texto).toMatchSnapshot();
    });

    test('TextoXSmall DEVE ser igual ao snapshot', () => {
      const texto = renderer
        .create(<S.TextoXSmall>Teste</S.TextoXSmall>)
        .toJSON();
      expect(texto).toMatchSnapshot();
    });

    test('TextoSmall DEVE ser igual ao snapshot', () => {
      const texto = renderer
        .create(<S.TextoSmall>Teste</S.TextoSmall>)
        .toJSON();
      expect(texto).toMatchSnapshot();
    });

    test('TextoMedium DEVE ser igual ao snapshot', () => {
      const texto = renderer
        .create(<S.TextoMedium>Teste</S.TextoMedium>)
        .toJSON();
      expect(texto).toMatchSnapshot();
    });
  });
});
