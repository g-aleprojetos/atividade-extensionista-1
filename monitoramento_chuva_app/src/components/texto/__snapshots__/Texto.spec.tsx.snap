// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Texto Renderização Texto2XSmall DEVE ser igual ao snapshot 1`] = `
<p
  className="sc-bcXHqe fXtsvG"
>
  Teste
</p>
`;

exports[`Texto Renderização Texto3XSmall DEVE ser igual ao snapshot 1`] = `
<p
  className="sc-bcXHqe lbnBAk"
>
  Teste
</p>
`;

exports[`Texto Renderização TextoMedium DEVE ser igual ao snapshot 1`] = `
<p
  className="sc-bcXHqe kaiKx"
>
  Teste
</p>
`;

exports[`Texto Renderização TextoSmall DEVE ser igual ao snapshot 1`] = `
<p
  className="sc-bcXHqe lltxVm"
>
  Teste
</p>
`;

exports[`Texto Renderização TextoXSmall DEVE ser igual ao snapshot 1`] = `
<p
  className="sc-bcXHqe cPPGUU"
>
  Teste
</p>
`;
